amazon-kinesis-client/src/test/java/software/amazon/kinesis/application/TestConsumer.java

```diff
@@ -115,9 +134,28 @@ private void cleanTestResources(StreamExistenceManager streamExistenceManager, L
     }
 
     private void startProducer() {
-        // Send dummy data to stream
-        this.producerExecutor = Executors.newSingleThreadScheduledExecutor();
-        this.producerFuture = producerExecutor.scheduleAtFixedRate(this::publishRecord, 60, 1, TimeUnit.SECONDS);
+        producerExecutor = Executors.newSingleThreadScheduledExecutor();
```

Comment:
This instantiation should occur outside this block, otherwise there's a non-zero risk that an Exception, or future refactor, might raise NPEs.